tofu/geom/_plot_optics.py

```diff
@@ -1142,12 +1147,47 @@ def CrystalBragg_plot_focal_error_summed(
         linewidths=1.,
     )
 
+    # computing detector with exact position of det_ref
+    if det_ref:
+        if not tangent_to_rowland:
+            detector_comp = cryst.get_detector_approx(
+                ddist=ddist0,
+                di=di0,
+                dj=dj0,
+                dtheta=dtheta0,
+                dpsi=dpsi0,
+                tilt=tilt0,
+                lamb=lamb,
+                use_non_parallelism=use_non_parallelism,
+                tangent_to_rowland=False,
+            )
+        else:
+            detector_comp = cryst.get_detector_approx(
+                ddist=ddist0,
+                di=di0,
+                dj=dj0,
+                dtheta=dtheta0,
+                dpsi=dpsi0-angle_nout,
+                tilt=tilt0,
+                lamb=lamb,
+                use_non_parallelism=use_non_parallelism,
+                tangent_to_rowland=True,
+            )
```

Comment:
Shorter formulation:

```
        psi0bis = float(psi0)
        if tangent_to_rowland:
            dpsi0bis = dpsi0 - angle_nout

        detector_comp = cryst.get_detector_approx(
            ddist=ddist0,
            di=di0,
            dj=dj0,
            dtheta=dtheta0,
            dpsi=dpsi0bis,
            tilt=tilt0,
            lamb=lamb,
            use_non_parallelism=use_non_parallelism,
            tangent_to_rowland=tangent_to_rowland,
        )
```

Reply:
I am not sure I understand what is the difference between `det_ref` and `detector_comp` ?

Why not use `det_ref` directly?

Was it for debugging ?

Reply:
The `_get_local_coordinates_of_det` method provides us the 3 translations (`ddist`, `di`, `dj`) and rotations (`dtheta`, `dpsi`, `tilt`) of the detector of reference `det_ref` which we want to characterize with these variables. 
So `detector_comp` is the detector plotted with the results of this method.
I choose the way to plot both `det_ref` and `detector_comp` to be sure they are overlapping perfectly.
Even if we know that the way to compute these variables is correct, this is an easy way to ensure the method is being correctly its job.
Is this answer good enough ?